quelpa.el

```diff
@@ -1853,14 +1866,24 @@ the global var `quelpa-upgrade-p' is set to nil."
     (let* ((quelpa-upgrade-p (if current-prefix-arg t quelpa-upgrade-p)) ;shadow `quelpa-upgrade-p'
            (quelpa-stable-p quelpa-stable-p) ;shadow `quelpa-stable-p'
            (quelpa-force-upgrade-p quelpa-force-upgrade-p) ;shadow `quelpa-force-upgrade-p'
-           (cache-item (if (symbolp arg) (list arg) arg)))
+           (cache-item (if (symbolp arg) (list arg) arg))
+           version)
       (quelpa-parse-plist plist)
       (quelpa-parse-stable cache-item)
-      (quelpa-package-install arg)
+      (setq version (quelpa-package-install arg))
+      (setq cache-item `(,(car cache-item) ,@(plist-put (cdr cache-item) :version version)))
       (quelpa-update-cache cache-item)))
   (quelpa-shutdown)
   (run-hooks 'quelpa-after-hook))
 
+;; compatibility
+(advice-add #'package-delete :after
```

Comment:
We should avoid adding mandatory, unconditional advice to important functions like this.  It would be better to only do this when something Quelpa-related is called.